src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/Hop.tsx

```diff
@@ -101,14 +103,25 @@ export const Hop = ({
       hopIndex,
     }
   }, [activeTradeId, hopIndex])
+  const rateHopExecutionMetadataFilter = useMemo(
+    () => ({
+      tradeId: initialActiveTradeId,
+      hopIndex,
+    }),
+    [hopIndex, initialActiveTradeId],
+  )
 
   const {
     state: hopExecutionState,
-    allowanceApproval,
     permit2,
     swap,
   } = useAppSelector(state => selectHopExecutionMetadata(state, hopExecutionMetadataFilter))
 
+  // Get allowance approval data from initial (rate) tradeId
```

Comment:
[suggestion]: Could be worth adding some detail as to why this is needed, as at first glance it's not obvious the active trade id changes after the final quote is fetched after the approval step is done and is stored under a different ID in redux

Reply:
gm https://github.com/shapeshift/web/pull/8222